src/script/script_system.cpp

```diff
@@ -73,137 +79,149 @@ namespace Lumix
+			{
+				return m_script_renamed;
+			}
+
+
 			void setScriptPath(Component cmp, const string& str) override
 			{
-				m_paths[cmp.index] = string(str.c_str(), m_allocator);
-				stopScript(cmp.index);
-				if (!runScript(cmp.index))
-				{
-					g_log_warning.log("script") << "Could not run script " << str;
-				}
+				Lumix::Path old_path = m_paths[cmp.index];
+				m_paths[cmp.index] = str.c_str();
+				m_script_renamed.invoke(old_path, m_paths[cmp.index]);
 			}
 
 
-			void getDll(const char* script_path, char* dll_path, char* full_path, int max_length)
+			virtual Component getNextScript(const Component& cmp) override
 			{
-				copyString(dll_path, max_length, script_path);
-				int32_t len = (int32_t)strlen(script_path);
-				if (len > 4)
+				for (int i = cmp.index + 1; i < m_script_entities.size(); ++i)
 				{
-					copyString(dll_path + len - 4, 5, ".dll");
+					if (m_script_entities[i] != -1)
+					{
+						return Component(Entity(&m_universe, m_script_entities[i]), SCRIPT_HASH, this, i);
+					}
 				}
-				copyString(full_path, max_length, m_engine.getBasePath());
-				catCString(full_path, max_length, "\\");
-				catCString(full_path, max_length, dll_path);
+				return Component::INVALID;
 			}
+			
 
+			virtual Component getFirstScript() override
+			{
+				for (int i = 0; i < m_script_entities.size(); ++i)
+				{
+					if (m_script_entities[i] != -1)
+					{
+						return Component(Entity(&m_universe, m_script_entities[i]), SCRIPT_HASH, this, i);
+					}
+				}
+				return Component::INVALID;
+			}
 
+			
 			void getScriptDefaultPath(Entity e, char* path, char* full_path, int length, const char* ext)
 			{
 				char tmp[30];
 				toCString(e.index, tmp, 30);
 
 				copyString(full_path, length, m_engine.getBasePath());
-				catCString(full_path, length, "\\scripts\\e");
+				catCString(full_path, length, "e");
 				catCString(full_path, length, tmp);
 				catCString(full_path, length, ".");
 				catCString(full_path, length, ext);
 
-				copyString(path, length, "scripts\\e");
+				copyString(path, length, "e");
 				catCString(path, length, tmp);
 				catCString(path, length, ".");
 				catCString(path, length, ext);
 			}
 
 
-			virtual void beforeScriptCompiled(const Lumix::Path& path) override
+			virtual void afterScriptCompiled() override
 			{
-				for (int i = 0; i < m_paths.size(); ++i)
+				if (!m_module)
 				{
-					if (m_paths[i] == path.c_str())
+					const char* library_path = "scripts/Debug/entities.dll";
+					m_module = LoadLibrary(library_path);
+					if (!m_module)
 					{
-						stopScript(i);
+						g_log_error.log("script") << "Could not load " << library_path;
+						return;
 					}
-				}
-			}
-
-
-			virtual void afterScriptCompiled(const Lumix::Path& path) override
-			{
-				for (int i = 0; i < m_paths.size(); ++i)
-				{
-					if (m_paths[i] == path.c_str())
+					m_update_function = (UpdateFunction)GetProcAddress(m_module, "update");
```

Comment:
This should be moved to some platform part. Some library loader or ...  
Some task to do it would be enough though.